### DISCUSSION TITLE
Detect in getServerSideProps if a request is the initial page request or a page transition request of a mounted page

### DISCUSSION THREAD

**Discussion by sebsauer90** [vercel/next.js#46308] [Help]
Hey everyone,

in our application we want to distinguish, if a request of `getServerSideProps` is the initial page (document) request, or a page transition request (JSON). For example, the first request should load user data on server side, but that should only be done once and not with each following page transition on pages with `getServerSideProps`.

Now we have seen that there is a new header `x-nextjs-matched-path` with next 13, which is only been set on page transition (JSON) requests. 

Do you think it could be an idea to distinguish an initial request and a page transition by reading this header?

Thanks in advance for your help and/or ideas :)

**Comment by icyJoseph** [comment]
This is a code/architecture smell... 

The "official" solution is to use App Directory with Server Components.

However, someone did trick it into going GSSP only on first load... and then on SPA transitions it does the thing client side:

- https://www.gregroz.me/article/nextjs-getServerSideProps-interception

At your own risk...

One thing I'd say is that the "first" request is made to the `/page` route, while the others are made to `/_next/data/page.json` sort of, the structure of the URL is irrelevant, but that's also one way you might be able to tell them apart. 

You'd need to check `context.req.url` and `context.resolvedUrl`.

I've never had the need to do this, so if there's any bugs waiting around the corner, I don't know...

**Reply by sebsauer90** [reply]
Hey [Joseph](https://github.com/icyJoseph),

first, thank you for you answer!

As long as server components are in beta, it isn't a solution for us, because we are already productive. And for now, i don't find a good way to use data which is fetched on server side, on client side.

Anyways, thank you very much for the hint with `context.req.url`. I think that could be a good solution for us! :)

Have a nice day!
